### PR TITLE
[csproj] Update generator to handle platform-specific assemblies

### PR DESCRIPTION
**Pull request by akoeplinger**
After https://github.com/mono/mono/pull/4837 was merged the csproj update bot started sending PRs which hardcoded net_4_x-linux as the OutputPath (see #4846).

Updated the csproj generator to detect this and use a variable in the OutputPath instead. It'll autodetect based on the current OS and can be overriden by passing `/p:HostPlatform=foobar` to msbuild.

Ideally we'd have configurations for each platform in the csproj since in the future the file lists will be different but this is a first step.

### CODE REVIEW

File: msvc/scripts/csproj.tmpl [hunk 1]
```diff
@@ -9,6 +9,9 @@
     <ProjectGuid>@PROJECTGUID@</ProjectGuid>
     <OutputType>@OUTPUTTYPE@</OutputType>
     <NoWarn>@DISABLEDWARNINGS@</NoWarn>
+    <HostPlatform Condition=" '$(HostPlatform)' == '' and '$(OS)' == 'Windows_NT'">win32</HostPlatform>
```
Comment (marek-safar) [inline msvc/scripts/csproj.tmpl]: Why do we need this?

Comment (akoeplinger) [inline msvc/scripts/csproj.tmpl]: This is to autodetect the current platform where msbuild is running on and use the corresponding configuration, basically like we do it in configure.ac. If you want to cross-compile you can override this by passing `/p:HostPlatform=darwin` to msbuild.

Comment (marek-safar) [inline msvc/scripts/csproj.tmpl]: I am not sure we need this. Where is HostPlatform used ?

Comment (akoeplinger) [inline msvc/scripts/csproj.tmpl]: Right now in the OutputPath (the property is written by the generator). In the future we can use it to select different files for different platforms.

**Pull request merged**